scheduler/workflow.go

```diff
@@ -431,7 +452,12 @@ func submitPublishJob(pj job, t *task, wg *sync.WaitGroup, pu *publishNode) {
 	// Decrement the waitgroup
 	defer wg.Done()
 	// Create a new process job
-	j := newPublishJob(pj, pu.Name(), pu.Version(), pu.InboundContentType, pu.config.Table(), t.metricsManager, t.id)
+	var j job
+	if pu.Hostname != "" {
+		j = newPublishJob(pj, pu.Name(), pu.Version(), pu.InboundContentType, pu.config.Table(), t.RemoteManagers.Get(pu.Hostname), t.id)
```

Comment:
Yet we impose that overhead for the REST API to talk to instances of scheduler and control that are local. The concept of local and remote is just going to clutter the code base with these constant checks if we should be talking to a remote instance or a local instance and trying to manage both versus just treating them all the same regardless if it's local or remote.

The only benefit to continue to allow talking in-process is giving the end user the ability to completely disable GRPC communications if they choose not to use distributed workflow with snap, which we currently do not give them the ability to do.

So in my opinion, we should either have everything communicate over RPC (local and remote) or keep everything local communicating over in-process, but we shouldn't mix and match.